src/viam/examples/modules/complex/test_complex_module.cpp

```diff
@@ -69,60 +70,25 @@ BOOST_AUTO_TEST_SUITE_END()
 
 BOOST_AUTO_TEST_SUITE(test_gizmo_client_server)
 
-// This sets up the following architecture
-// -- MockComponent
-//        /\
-//
-//        | (function calls)
-//
-//        \/
-// -- ComponentServer (Real)
-//        /\
-//
-//        | (grpc InProcessChannel)
-//
-//        \/
-// -- ComponentClient (Real)
-//
-// This is as close to a real setup as we can get
-// without starting another process
-//
-// The passed in lambda function has access to the ComponentClient
-template <typename Lambda>
-void gizmo_server_to_mock_pipeline(Lambda&& func) {
-    GizmoServer gizmo_server;
-    gizmo_server.resource_manager()->add(std::string("testgizmo"), test_gizmo_impl::get_gizmo());
-
-    grpc::ServerBuilder builder;
-    builder.RegisterService(&gizmo_server);
-
-    std::unique_ptr<grpc::Server> server = builder.BuildAndStart();
-
-    grpc::ChannelArguments args;
-    auto grpc_channel = server->InProcessChannel(args);
-    GizmoClient client("testgizmo", grpc_channel);
-    // Run the passed test on the created stack
-    std::forward<Lambda>(func)(client);
-    // shutdown afterwards
-    server->Shutdown();
-}
-
 BOOST_AUTO_TEST_CASE(test_do_one) {
-    gizmo_server_to_mock_pipeline([](Gizmo& client) -> void {
+    const auto mock = test_gizmo_impl::get_gizmo();
```

Comment:
I think there is a nice trick you can use to save some typing:

Up where you had `gizmo_server_to_mock_pipeline` before, do:

```
template<typename Lambda>
using gizmo_server_to_mock_pipeline = client_to_mock_pipeline<GizmoClient, GizmoServer, Lambda>;
```

Then you don't need to re-iterate the template parameters for the client and server everywhere. Maybe mod a rename if you don't like the name `gizmo_server_to_mock_pipeline` and want to call it `gizmo_client_to_mock_pipeline` or something.

Reply:
Spoke a bit offline; this would be cool, but it doesn't fully work as written there. `client_to_mock_pipeline` is a function not a type, so it can't be on the right hand side of a `using` statement. I may just leave the unfortunately verbose `client_to_mock_pipeline<*Client, *Server>` pattern around.